tests/rptest/tests/upgrade_test.py

```diff
@@ -614,3 +614,37 @@ def cloud_log_size() -> int:
         # Assert manifest has not been mutated by comparing sizes
         total_cloud_size_after_upgrade = cloud_log_size()
         assert total_cloud_size_before_upgrade <= total_cloud_size_after_upgrade, f"Mismatch in cloud storage size after upgrade, bytes before: {total_cloud_size_before_upgrade} bytes_after: {total_cloud_size_after_upgrade}"
+
+
+class InstallSelfTest(RedpandaTest):
+    @cluster(num_nodes=1)
+    def test_redpandainstaller_install(self):
+        """
+        Smoke test that checks RedpandaInstaller.install
+        :return:
+        """
+        install = lambda version: self.redpanda._installer.install(
+            self.redpanda.nodes, version)
+        start = lambda: self.redpanda.start(self.redpanda.nodes[0:1])
```

Comment:
nit: just `self.redpanda`?

Also could you explicitly pass in `clean_nodes=True`, since this test depends on it?

Reply:
start is used more than once, so probably only want clean_nodes on the first one?  Or possibly not at all, as this is a RedpandaTest that should get its first startup (the one where clean is needed) automatically in setUp.

I'm not sure the first call to start() makes sense below on line 635, because isn't the cluster already started at this point?

If you want to delay redpanda startup until your first explicit start, then you would override setUp() with a `pass` and then use self.redpanda.start(clean_nodes=True) on your first startup of nodes.

Reply:
oh, now I get it.  It's intentionally nuking redpanda each time.